src/tools/pr_8799/src/virtcontainers.rs

```diff
@@ -0,0 +1,109 @@
+use anyhow::{anyhow, Result};
+use std::{collections::HashMap, fs, path::Path};
+
+// Create a map of tids-vcpus from /proc/pid/task subdirectories
+pub fn get_ch_vcpu_tids_by_path(proc_path: &str) -> Result<HashMap<u32, u32>> {
+    let mut tids = HashMap::new();
+
+    if proc_path.trim().is_empty() {
+        return Err(anyhow!("Path must be non-empty strings."));
+    }
+
+    let src = std::fs::canonicalize(proc_path)
+        .map_err(|e| anyhow!("Invalid /proc/pid path: {proc_path}: {e}"))?;
```

Comment:
Nit: Since this functions accepts a path, it would be better to say "proc path" I think (here and elsewhere):

```suggestion
        .map_err(|e| anyhow!("Invalid proc path: {proc_path}: {e}"))?;
```